integrated_channels/canvas/client.py

```diff
@@ -265,7 +276,114 @@ def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-
         # Todo: There isn't a great way for users to delete course completion data
         pass
 
+    def cleanup_duplicate_assignment_records(self, courses):
```

Comment:
I'd break this down a bit further with some _ sub-functions and unit test ability: a 100+ line function with a while loop nested in a for loop is hard to read.

Reply:
+1!